paasta_tools/contrib/reset_kubernetes_cluster.py

```diff
@@ -0,0 +1,219 @@
+        print(f"\t---command output ({command_string})---")
+        print(indent(result.stdout.decode(), "\t\t"))
+        print("\t---end command output---")
+
+    def run(self) -> None:
+        """Actually reset the cluster"""
+        # Stop kubernetes components on all master
+        print("> Disabling puppet on kube masters...")
+        self.cmd(
+            self.master_node_names + self.other_node_names,
+            'toggle-puppet disable --reason "Resetting cluster"',
+        )
+        print("> Disable monitoring...")
+        self.cmd(
+            self.master_node_names + self.other_node_names,
+            'downtime $(hostname -f) 3h "Resetting cluster"',
+        )
+        print("> Stopping kubelet on masters...")
+        self.cmd(self.master_node_names, "sudo systemctl stop kubelet")
+        print("> Killing all containers...")
+        self.cmd(self.master_node_names, "sudo docker kill $(sudo docker ps -q)")
+        print("> Pruning all containers...")
+        self.cmd(self.master_node_names, "sudo docker container prune -f")
+        print("> Removing kubernetes_bootstrap_done.txt...")
+        self.cmd(
+            self.master_node_names,
+            "sudo rm -f /etc/facter/facts.d/kubernetes_bootstrap_done.txt",
+        )
+        print("> Running kubeadm reset, this may take a while...")
+        self.cmd(self.master_node_names, "sudo kubeadm reset -f")
+
+        print("> Removing Puppet resource state in S3")
+        self.cmd(
+            [self.master_node_names[0]],
+            (
+                "aws s3 rm --recursive s3://yelp-kubernetes-puppet-resources-"
+                f"state-{self.cluster_ecosystem}/{self.cluster_name}/"
+            ),
+        )
+
+        print(
+            f"> Deleting etcd prefix '/{self.etcd_prefix}' on host '{self.etcd_host}'"
+        )
+        self.cmd(
+            [self.etcd_host], f"sudo etcdctl del '/{self.etcd_prefix}' --from-key=true"
+        )
+
+        print(f"> Running puppet (on {self.master_node_names[0]} node)")
+        self.cmd([self.master_node_names[0]], "sudo run-puppet -f")
+        print(f"> Running kube-init (on {self.master_node_names[0]} node)")
+        # # -H is required to set the home directory to the one of root
+        # # which has the .dockercfg file to pull k8s images
+        self.cmd([self.master_node_names[0]], "sudo -H /nail/sys/bin/kube-init")
+        print("> Running puppet (on all nodes)")
+        self.cmd(
+            self.master_node_names, "sudo run-puppet -f",
+        )
+        print("> Running kube-init")
+        self.cmd(self.master_node_names, "sudo -H /nail/sys/bin/kube-init")
+        print("> Untainting master nodes...")
+        for host in self.master_node_names:
+            self.cmd(
+                [host],
+                (
+                    "sudo KUBECONFIG=/etc/kubernetes/admin.conf kubectl taint nodes "
+                    f"{host} node-role.kubernetes.io/master:NoSchedule-"
```

Comment:
Also including my comment from the internal PR:

```
I would like to avoid taking dependencies on the admin.conf (and therefore, the generic administrative client certificate) for pretty much anything. If humans are expected to use this, I'd prefer that you use your own okta federated credentials for authentication. If this is intended to be run as a service or script, then you need to create a new set of credentials for it, ideally scoped only to the actions and resources it needs access to (which would probably alleviate some of the other aforementioned concerns about accidentally running this on the wrong cluster)
```